packages/mui-material/src/utils/getChildRef.js

```diff
@@ -1,8 +1,16 @@
 import * as React from 'react';
 
 export default function getChildRef(child) {
-  if (!child || !React.isValidElement(child)) {
-    return null;
+  if (process.env.NODE_ENV !== 'production') {
+    if (!React.isValidElement(child)) {
+      console.error(
+        [
+          "MUI: getChildRef doesn't accept array as a child.",
+          'Consider providing a single element instead.',
+        ].join('\n'),
+      );
+      return null;
```

Comment:
What's the reasoning to return `null` in non-prod envs? I think it's probably better to have a hard error.

Reply:
We can't have code branching between dev and production
```suggestion
```

Reply:
Yeah, I meant let the flow continue and let the function error naturally if the param is incorrect.